File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/cpp/F14.P2.cpp
#include <cstdio>
#include <variant>

struct Circle { double r; };
struct Rect { double w, h; };
using Shape = std::variant<Circle, Rect>;

double area_of(Shape s) {
// BEGIN PROBE F14.P2
    struct Area {
        double operator()(Circle c) const { return 3.141592653589793 * c.r * c.r; }
        double operator()(Rect r) const { return r.w * r.h; }
    };
    double area = std::visit(Area{}, s);
    return area;
// END PROBE F14.P2
}

int main() {
    std::printf("%.6f\n", area_of(Circle{2.0}));
}

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/swift/F14.P2.swift
enum Shape {
    case circle(r: Double)
    case rect(w: Double, h: Double)
}

func areaOf(_ s: Shape) -> Double {
// BEGIN PROBE F14.P2
let area: Double = switch s {
case let .circle(r): 3.141592653589793 * r * r
case let .rect(w, h): w * h
}
return area
// END PROBE F14.P2
}

print(areaOf(.circle(r: 2.0)))
